packages/blade/src/components/Tabs/Tabs.native.tsx

```diff
@@ -0,0 +1,214 @@
+
+  return tabList.props.children;
+};
+
+const getTabPanels = (children: React.ReactNode): Record<string, () => React.ReactElement> => {
+  const childs = React.Children.toArray(children) as React.ReactElement[];
+  return childs
+    .filter((child) => getComponentId(child) === 'TabPanel')
+    .reduce((prev, curr) => {
+      return { ...prev, [curr.props.value]: () => curr.props.children };
+    }, {});
+};
+
+const getRoutes = (tabs: React.ReactElement[]) => {
+  return tabs.map((TabComponent, index) => ({
+    index,
+    title: TabComponent.props.children,
+    key: TabComponent.props.value,
+    value: TabComponent.props.value,
+    leading: TabComponent.props.leading,
+    trailing: TabComponent.props.trailing,
+  }));
+};
+
+type Route = {
+  value?: string;
+  index?: number;
+};
+
+const getRouteIndexFromValue = ({ value, routes }: { value?: string; routes: Route[] }) =>
+  routes.findIndex((route) => route.value === value);
+
+const getRouteValueFromIndex = ({ index, routes }: { index?: number; routes: Route[] }) =>
+  routes.find((route) => route.index === index)?.value!;
+
+const Tabs = ({
+  children,
+  defaultValue,
+  value,
+  onChange,
+  size = 'medium',
+  variant = 'bordered',
+  autoWidth = false,
+}: TabsProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const isFirstRender = useFirstRender();
+  const tabs = getTabs(children);
+  const panels = getTabPanels(children);
+  const routes = getRoutes(tabs);
+  const isFilled = variant === 'filled';
+
+  const [selectedValue, setSelectedValue] = useControllableState({
+    defaultValue,
+    value,
+    onChange: (value) => {
+      if (isFirstRender) return;
+      onChange?.(value);
+    },
+  });
+
+  const index = getRouteIndexFromValue({
+    value: selectedValue,
+    routes,
+  });
+
+  const setIndex = React.useCallback(
+    (index: number) => {
+      const value = getRouteValueFromIndex({ index, routes });
+      setSelectedValue(() => value);
+    },
+    [routes, setSelectedValue],
+  );
+
+  const isVertical = false;
+  const contextValue = {
+    baseId: '',
+    selectedValue,
+    isVertical,
+    size,
+    variant,
+    autoWidth,
+    setSelectedValue: setIndex,
+  };
+
+  // Set initial value
+  React.useLayoutEffect(() => {
+    if (selectedValue) return;
+    setIndex(0);
+  }, [selectedValue, setIndex]);
+
+  const renderTabLabel = React.useCallback(
+    ({ route, focused }) => {
+      const { title, leading, trailing } = route;
+      const selectedState = focused ? 'selected' : 'unselected';
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const validatedTrailingComponent = useTabsItemPropRestriction(trailing, size);
+
+      return (
+        <StyledTabButton autoWidth={!autoWidth && !isFilled} variant={variant} size={size}>
+          <Box display="flex" alignItems="center" flexDirection="row" gap="spacing.3">
+            {leading
+              ? React.cloneElement(leading as React.ReactElement, {
```

Comment:
What are we doing here? Cloning I'm assuming an Icon element and adding size & color?

Why this way? See how we're doing it in other places like List. We should have a different component like `TabIcon` which only has limited props. Cloning and adding sizes this way will lead to confusion because users might pass a size and then when it doesn't work they will raise an issue.

Reply:
> Cloning and adding sizes this way will lead to confusion

I've already modified this to not use cloneElement. 


> We should have a different component like TabIcon which only has limited props.

No, we are already doing this this way in newer components like BottomSheet, BaseHeader etc. Introducing new component for every single Parent component is not very ideal. 

Since we override both color/size prop in icon, consumers will simply use it like so: 

```
<TabItem icon={InfoIcon} />
```

No need to introduce new TabItemIcon component.